File: cpp/leetcode/075.cpp
class Solution {
public:
    void sortColors(vector<int> &nums) {
        if (nums.empty()) {
            return;
        }
        int left = 0, right = nums.size() - 1, i = 0;
        while (i <= right) {
            if (nums[i] == 0) {
                swap(nums[left++], nums[i++]);
            } else if (nums[i] == 2) {
                swap(nums[right--], nums[i]);
            } else {
                i++;
            }
        }
    }
};

File: python/leetcode/075.py
class Solution:
    def sortColors(self, num):
        if not num:
            return
        l, r, i = 0, len(num) - 1, 0
        while i <= r:
            if num[i] == 2:
                num[i], num[r] = num[r], num[i]
                r -= 1
            elif num[i] == 0:
                num[i], num[l] = num[l], num[i]
                l += 1
                i += 1
            else:
                i += 1
